cluster/plan.go

```diff
@@ -558,6 +560,26 @@ func (c *Cluster) BuildKubeletProcess(host *hosts.Host, serviceOptions v3.Kubern
 			fmt.Sprintf("%s:c:/host/run", path.Join(host.PrefixPath, "/run")),
 		}...)
 	} else {
+		parsedVersion, err := getClusterVersion(c.Version)
+		if err != nil {
+			logrus.Debugf("Error while parsing cluster version: %s", err)
+		}
+		parsedJanK8s123Version, err := getClusterVersion(JanK8s123Version)
```

Comment:
I'm missing context about the discussion with @kinarashah referenced [here](https://github.com/rancher/rke/pull/3169#issuecomment-1445654373), but can we actually include `v1.23.16-rancher1` here as it's already released? (Actual released version is `v1.23.16-rancher2-1` btw)

Note that we're planning to add `v1.23.16-rancher2-2` per https://github.com/rancher/kontainer-driver-metadata/pull/1110, so not sure if we can use that instead for checks.